### PR TITLE
Add release post for 3.34 (2021-11-23)

### PR DESCRIPTION
**Pull request by iskyOS**
https://handbook.sourcegraph.com/marketing/product-marketing/release_post_process

### CODE REVIEW

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:20]:
```suggestion
In this release, we are introducing server-side Batch Changes. Computing batch changes can now be offloaded to [executors](https://docs.sourcegraph.com/admin/deploy_executors), allowing you to:
```

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:18]:
```suggestion
By default, Batch Changes uses a command line interface in your local environment to compute batch changes. This can be impractical for creating batch changes affecting hundreds or thousands of repositories or if the batch change steps require CPU, memory, or disk resources that are not available locally.
```

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:22]:
```suggestion
- Run large-scale batch changes that would be impractical to compute locally
```

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:23]:
```suggestion
- Speed up batch change creation time by distributing batch change computing over several executors
```

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:24]:
```suggestion
- Reduce the setup time required to onboard new users to Batch Changes
```

**Comment by jyavorska** [comment]
Reviewed for Product (on behalf of @christinaforney) this morning, though either content is very light or folks are late adding it. I'll keep an eye on it and will review if more content comes in.

I've also pinged Code Graph PMs to double check there isn't more.

**Comment by jyavorska** [comment]
Added changelog

File: blogposts/2021/release-3.34
Comment (macraig) [inline blogposts/2021/release-3.34:19]:
```suggestion
## Find Implementations for Go
Navigating to implementations is now available for Go! If you have precise code intelligence enabled in your Go repositories, you can now click on Find Implementations to navigate to a symbol's interface definition. If you're at the interface definition itself, clicking on Find Implementations will show all the places where the interface is being implemented, allowing you to explore how it's being used by other users across repositories. It will also show you which interfaces a struct implements. 
<img width="1193" alt="Screenshot 2021-11-22 at 18 28 39" src="https://user-images.githubusercontent.com/1657213/142938393-7aed0c41-28b8-4cab-bf0d-2f9fd7f1078c.png">
<img width="1193" alt="Screenshot 2021-11-22 at 18 29 03" src="https://user-images.githubusercontent.com/1657213/142938404-3b845bec-5da2-4ec7-bf17-0e273ff38dcd.png">
```

File: blogposts/2021/release-3.34
Comment (macraig) [inline blogposts/2021/release-3.34:23]:
```suggestion

# 3.34.0
```

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:17]:
```suggestion
When your search comes back with no results it can sometimes be hard to know why. We now display helpful tips on how to use search, examples of how to solve common query issues, and links to help videos addressing common use cases.
```

**Comment by jyavorska** [comment]
@iskyOS I did another edit pass with the new items, and it looks good to me.

**Comment by jyavorska** [comment]
@iskyOS is there a way to see this in the review app? It doesn't seem to be at https://deploy-preview-4993--sourcegraph.netlify.app/blog.

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:17]:
```suggestion
  - description: "To help site admins debug issues with webhook delivery we have added logging of incoming Batch Changes webhooks. By default, sites without encryption will log webhooks for three days and sites with encryption will not log webhooks."
```

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:20]:
```suggestion
  - description: "In order to allow writing to the `.github` directory in repositories, Batch Changes now requests the `workflow` scope on GitHub personal access tokens. If you have already configured a GitHub PAT for Batch Changes, we suggest adding this scope."
```

@malomarrec I think this would also benefit from explicitly saying why you'd want to allow writing to the `.github` folder.

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:23]:
```suggestion
  - description: "In order to increase search speed, the search UI's repository count and the GraphQL API's `search().repositories`/`search().repositoriesCount` have changed semantics from being the set of searchable repositories to the set of repositories with matches. In a future release, we'll introduce separate fields for searchable repositories."
```

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:37]:
```suggestion
Searches that return no results can result in a confusing experience. When this occurs, we now display helpful tips on how to use search, examples of how to solve common query issues, and links to tutorial videos addressing common use cases.
```

File: blogposts/2021/release-3.34
Comment (jyavorska) [inline blogposts/2021/release-3.34:41]:
```suggestion
If you have precise code intelligence enabled for your Go repositories, you can now click on "Find Implementations" to navigate to a symbol's interface definition. If you're at the interface definition itself, clicking on the same button will show all the places where the interface is being implemented, allowing you to explore how it's being used by other users across repositories. It can also show you which interfaces a struct implements.
```

**Review by jyavorska — APPROVED** [review]
@iskyOS I've left some suggestions as I wasn't sure if I should just make them or send them to you. If you agree (or if I'm supposed to just do it myself) feel free to batch commit before merging. Thanks!

File: blogposts/2021/release-3.34
Comment (jyavorska) [reply blogposts/2021/release-3.34:11]: cc @benvenker I added a bit more on why this is valuable.

File: blogposts/2021/release-3.34
Comment (jyavorska) [reply blogposts/2021/release-3.34:14]: cc @lguychard updated this to lead with the benefit.

File: blogposts/2021/release-3.34
Comment (jyavorska) [reply blogposts/2021/release-3.34:17]: cc @malomarrec updated this to lead with the benefit.

File: blogposts/2021/release-3.34
Comment (jyavorska) [reply blogposts/2021/release-3.34:23]: cc @lguychard updated this to lead with the benefit.

File: blogposts/2021/release-3.34
Comment (jyavorska) [reply blogposts/2021/release-3.34:37]: cc @benvenker just minor edits here to edit it down. The removed terms weren't adding a lot of meaning.

File: blogposts/2021/release-3.34
Comment (jyavorska) [reply blogposts/2021/release-3.34:41]: cc @macraig updated this to lead with the benefit.

File: blogposts/2021/release-3.34
Comment (lguychard) [reply blogposts/2021/release-3.34:14]:
```suggestion
  - description: "In order to reduce load on the Gitserver service on installations with a large number of repositories, the search indexer only polls repositories that have been marked as changed. If you notice index staleness you can try disabling by setting the environment variable `SRC_SEARCH_INDEXER_EFFICIENT_POLLING_DISABLED` on `sourcegraph-frontend`."
```

Thanks! Slight update to put "large number of repos" back in, also clarify that this is about 
 reducing load on Sourcegraph's Gitserver service.

File: blogposts/2021/release-3.34
Comment (lguychard) [reply blogposts/2021/release-3.34:23]: Thanks, LGTM!

File: blogposts/2021/release-3.34
Comment (rebeccadee) [inline blogposts/2021/release-3.34:7]:
```suggestion
published: true
```

**Comment by rebeccadee** [comment]
@jyavorska For the preview to be available `published: true` needs to be set in the front matter, and also the file was missing the .md extension. It's working now!

> @iskyOS is there a way to see this in the review app? It doesn't seem to be at https://deploy-preview-4993--sourcegraph.netlify.app/blog.

File: blogposts/2021/release-3.34.md
Comment (rebeccadee) [inline blogposts/2021/release-3.34.md:39]: Is there some other way we can format the images? They're a little tough on the eyes at the moment with no line breaks between them and the text, and no drop shadow. I also think it'd be worthwhile to add captions so people understand what they're seeing. I tried to use the basic markdown formatting to see if that helps but that makes the images enormous 😅 @jyavorska you have helped me with rogue images in the handbook, wondering if you have ideas here?

**Review by rebeccadee — COMMENTED** [review]
@iskyOS LGTM, though I've left one comment about the images which aren't rendering well at the moment!

File: blogposts/2021/release-3.34.md
Comment (jyavorska) [reply blogposts/2021/release-3.34.md]: We can definitely change the embedded style here. It looks like bootstrap is available, so I will use some basic styling.

Comment (jyavorska) [reply blogposts/2021/release-3.34.md]: Ok it should be a bit better now. I don't know where people were getting the idea to add the `style` attribute but I found it in some recent ones too and updated those to use the new `blog-images` css class. If we figure out where that is coming from, we can update the instructions for everyone to use `blog-images` instead.

File: blogposts/2021/release-3.34.md
Comment (jyavorska) [inline blogposts/2021/release-3.34.md:44]: @macraig these images are showing very nearly the exact same thing as best I can tell, what's the difference between the two?

File: blogposts/2021/release-3.34.md
Comment (jyavorska) [inline blogposts/2021/release-3.34.md:39]:
```suggestion
<img class="blog-image" title="No results page" alt="This image shows the new no results page, with resources available to troubleshoot your search when no results are returned." src="https://storage.googleapis.com/sourcegraph-assets/blog/3.34/no_results_page.png">
```

cc @benvenker the alt and title tags here seemed wrong so I updated them. If these are wrong too please follow up with a correction.

File: blogposts/2021/release-3.34.md
Comment (jyavorska) [reply blogposts/2021/release-3.34.md]: @iskyOS I wouldn't block merging the release post on this, a fix could come in a follow up (or not).

Comment (macraig) [reply blogposts/2021/release-3.34.md]: @jyavorska fair point. One shows where an interface is implemented across many different repositories, the other shows how you see where the interface is defined from a repo that implements it. I understand this seems too similar and confusing (and we're actually currently looking into a way of making these two directions clearer in the UI). I think we can remove the second image, I'll do that right now.

File: blogposts/2021/release-3.34.md
Comment (macraig) [inline blogposts/2021/release-3.34.md:45]:
```suggestion
```

**Pull request merged**
